system-functionalities.md

```diff
@@ -360,13 +363,15 @@ In [remote-only secret generation protocol](#remote-only-secret-generation-and-s
 - In [3.iii](#remotegen-client-output), the client outputs both `key_id` and `public_key` to the calling application.
 
 ### Importing a Signing Key
-Both versions of key import should return the public component of the generated key to the calling application.
+Both versions of key import are modified to return the public component of the imported key to the calling application.
 
 In [local import with remote backup](#local-import-with-remote-backup):
```

Comment:
Question: why not assume you're importing a full keypair?

Reply:
In the implementation, the `SigningKeyPair` type is maybe inappropriately named -- it actually only holds the secret bytes and regenerates the public key on demand. When I actually instantiate the type, it only expects the private key bytes. We could always take both bits of the keypair and validate them against each other, but it didn't seem like a strong security benefit.

I also think key formats usually have separate public key / private key files and don't squash them together into one (actually, I went on a long tangent and according to the [PEM RFC](https://www.rfc-editor.org/rfc/rfc7468), you MAY put multiple encodings into one file, but I'm pretty sure OpenSSL defaults to putting them separately).

Reply:
I'm OK with following OpenSSL defaults. I don't have a strong opinion on this, was just curious.